meson.build

```diff
@@ -0,0 +1,222 @@
+        '-Wredundant-decls',
+        '-Wno-missing-field-initializers'
+    ]
+    add_project_arguments(cc.get_supported_arguments(cflags), language: 'c')
+endif
+link_args = cc.get_supported_link_arguments(['-prefer-non-pic'])
+
+check_functions = [
+    'strdup',
+    'strndup'
+]
+
+m_dep = cc.find_library('m', required: false)
+if m_dep.found()
+    deps += m_dep
+endif
+
+iconv_dep = cc.find_library('iconv', required: false)
+if iconv_dep.found()
+    deps += iconv_dep
+endif
+if iconv_dep.found() or cc.has_function('iconv_open')
+    conf.set('CONFIG_ICONV', 1)
+endif
+
+# ASM
+nasm_args = []
+nasm = find_program('nasm', required: get_option('asm'))
+enable_asm = nasm.found()
+
+if enable_asm
+    # NASM version check
+    nasm_r = run_command(nasm, '-v')
+    out = nasm_r.stdout().strip().split()
+    if out[1].to_lower() == 'version'
+        if out[2].version_compare('< 2.10')
+            warning('nasm is too old (found @0@); ASM functions are disabled.'.format(out[2]))
+            enable_asm = false
+        endif
+    else
+        warning('Unexpected nasm version string: @0@; disabling.'.format(nasm_r.stdout()))
+        enable_asm = false
+    endif
+endif
+
+# Architecture check
+if host_machine.cpu_family() == 'x86'
+    bittype = '32'
+    nasm_args += '-DARCH_X86_64=0'
+    intel = true
+elif host_machine.cpu_family() == 'x86_64'
+    bittype = '64'
+    nasm_args += ['-DARCH_X86_64=1', '-DPIC']
+    intel = true
+else
+    intel = false
+endif
+
+if enable_asm and not intel
+    warning('ASM functions are not available on non-intel machines; disabling.')
+    enable_asm = false
+elif (not enable_asm) and intel
+    warning('Install nasm-2.10 or later for a significantly faster libass build.')
+endif
+
+if enable_asm
+    conf.set('CONFIG_ASM', 1)
+
+    if host_machine.system() == 'windows'
+        nasm_args += ['-f', 'win' + bittype]
+        if bittype == '64'
+            nasm_args += ['-DHAVE_ALIGNED_STACK=1']
+        else
+            nasm_args += ['-DHAVE_ALIGNED_STACK=0', '-DPREFIX']
+        endif
+    elif host_machine.system() == 'darwin'
+        nasm_args += ['-f', 'macho' + bittype, '-DPREFIX', '-DHAVE_ALIGNED_STACK=1']
+    else
+        nasm_args += ['-f', 'elf' + bittype, '-DHAVE_ALIGNED_STACK=1']
+    endif
+
+    nasm_args += ['-DHAVE_CPUNOP=0', '-Dprivate_prefix=ass']
+    nasm_args += ['-I', meson.current_source_dir() / 'libass' + '/', '-o', '@OUTPUT@', '@INPUT@']
+
+    nasm_gen = generator(nasm, output: '@BASENAME@.o', arguments: nasm_args)
+endif
+
+# Misc deps
+freetype_dep = dependency('freetype', version: '>= 2.2.1',
+                          default_options: ['default_library=' + get_option('default_library'), # workaround for https://github.com/mesonbuild/meson/issues/8047
+                                            'zlib=builtin',
+                                            'png=enabled'])
+if freetype_dep.found()
+    deps += freetype_dep
+    conf.set('CONFIG_FREETYPE', 1)
+endif
```

Comment:
Discussed on IRC, but I'll try to find history on the different freetype versions and why I ran into issues with that. It might be easiest to go back to not specifying a version...

Reply:
Reworked the logic a bit. Hopefully this works.